src/libraries/System.Data.Common/src/System/Data/Common/DbDataReaderExtensions.cs

```diff
@@ -49,8 +49,7 @@ private void PopulateFields()
                 DataTypeName = GetDbColumnValue<string?>("DataTypeName");
             }
 
-            // The following may return null, but local methods can't be annotated for that yet ([MaybeNull])
-            private T GetDbColumnValue<T>(string columnName) => _schemaColumns.Contains(columnName) && _schemaRow[columnName] is T value ? value : default!;
+            private T? GetDbColumnValue<T>(string columnName) => _schemaColumns.Contains(columnName) && _schemaRow[columnName] is T value ? value : default!;
```

Comment:
Do we still need `default!`? 
```suggestion
            private T? GetDbColumnValue<T>(string columnName) => _schemaColumns.Contains(columnName) && _schemaRow[columnName] is T value ? value : default;
```